src/Evolution/Executables/Burgers/EvolveBurgers.hpp

```diff
@@ -158,6 +159,9 @@ struct EvolutionMetavars {
   using observed_reduction_data_tags = observers::collect_reduction_data_tags<
       typename Event<events>::creatable_classes>;
 
+  using factory_classes = tmpl::map<
```

Comment:
I suggest to group this in a struct that points to some documentation, like the snippet I suggested above: https://github.com/sxs-collaboration/spectre/pull/3084#discussion_r611055276

Reply:
Agreed, sorry I didn't mean to suggest putting this documentation in all metavars. I'm suggesting to add a protocol with the documentation (e.g. in `Options/Protocols/FactoryCreation.hpp`) like in the snippet above https://github.com/sxs-collaboration/spectre/pull/3084#discussion_r611055276. This way the code in the metavars links directly to the documentation. Only secondarily the protocol could be used to validate the interface and maybe throw compiler errors early.

These are two reasons why I think we should move more towards documenting these things through protocols (or empty base classes, but protocols seem to fit the bill quite well):
1. Dev guides often don't get written at all, or get written later so people working with the new code have to figure out the interface by themselves, or become out-of-date because the code they describe changes. It's easier to write the docs for an interface and its intended use right when designing it. This is what you do when designing a base class, so I think this is also what we should do when designing a compile-time interface.
2. The documentation needs to be discoverable from the code it documents. There should be "breadcrumbs" leading you to the docs. If you don't understand the interface that a derived class implements, you follow its inheritance hierarchy up to the base class that documents the interface. Protocols follow the same idea: The code that implements a compile-time interface links directly to the protocol that defines and documents the interface.